sympy/geometry/point.py

```diff
@@ -234,6 +256,95 @@ def ambient_dimension(self):
         will be n"""
         return len(self)
 
+    @property
+    def orthogonal_direction(self):
+        """Returns a non-zero point in a direction
+        orthogonal to `self`."""
+
+        dim = self.ambient_dimension
+        if dim == 1 and self[0] != S.Zero:
+            raise ValueError("No orthogonal direction")
+
+        # if a coordinate is zero, we can put a 1 there and zeros elsewhere
+        if self[0] == S.Zero:
+            return Point([1] + (dim - 1)*[0])
+        if self[1] == S.Zero:
+            return Point([0,1] + (dim - 2)*[0])
+        # if the first two coordinates aren't zero, we can create a non-zero
+        # orthogonal vector by swapping them, negating one, and padding with zeros
+        return Point([-self[1], self[0]] + (dim - 2)*[0])
+
+    @staticmethod
+    def affine_rank(*points):
+        """The affine rank of a set of points is the dimension
+        of the smallest affine space containing all the points.
+        For example, if the points lie on a line (and are not all
+        the same) their affine rank is 1.  If the points lie on a plane
+        but not a line, their affine rank is 2."""
+
+        if len(points) == 0:
+            raise TypeError("Must have at least one point")
+
+        # make sure we're genuinely points
+        # and translate to every point to the origin
+        # XXX: issue #11238 we need to pre-simplify otherwise
```

Comment:
preceding lines:

# make sure all points are Points
# and translate all points to the origin

Regarding the 11238 hacks: could you add a test in the geometry suite which, if it passes, directs someone to go back through geometry and make the appropriate modifications? In case that isn't you, it would be good to include the post-fix line(s) as one or more comments and indicate which line(s) they are replacing.